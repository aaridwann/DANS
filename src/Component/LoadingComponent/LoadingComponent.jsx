import React from "react";
import { useSelector } from "react-redux";
import loading from "../../assets/loading.gif";

const _renderComponent = () => {
  return (
    <div className="absolute flex flex-col z-[500] items-center mx-auto justify-center bg-white/50 w-screen h-screen p-4">
      <img className=" w-20 h-auto mb-4" src={loading} />
      <h1 className=" text-2xl text-cyan-600 shadow-cyan-800  drop-shadow-2xl">
        LOADING....
      </h1>
    </div>
  );
};
function LoadingComponent() {
  const { loading } = useSelector((state) => state);
  const { LoadingState } = loading;
  const _renderLoading = LoadingState && _renderComponent();
  return _renderLoading;
}

export default LoadingComponent;
