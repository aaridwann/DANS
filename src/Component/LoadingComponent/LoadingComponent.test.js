import React from "react";
import { render } from "@testing-library/react";
import { useSelector } from "react-redux";
import LoadingComponent from "./LoadingComponent";
import "@testing-library/jest-dom/extend-expect";

jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
}));

describe("LoadingComponent", () => {
  beforeEach(() => {
    useSelector.mockReturnValue({
      loading: {
        LoadingState: true,
      },
    });
  });

  afterEach(() => {
    useSelector.mockClear();
  });

  test("renders correctly when loading state is true", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        loading: {
          LoadingState: true,
        },
      })
    );
    const { container } = render(<LoadingComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("does not render when loading state is false", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        loading: {
          LoadingState: false,
        },
      })
    );
    const { container } = render(<LoadingComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render LoadingComponent when LoadingState is true", () => {
    const { queryByText } = render(<LoadingComponent />);
    expect(queryByText("LOADING....")).toBeInTheDocument();
  });

  it("should not render LoadingComponent when LoadingState is false", () => {
    useSelector.mockReturnValue({
      loading: {
        LoadingState: false,
      },
    });
    const { queryByText } = render(<LoadingComponent />);
    expect(queryByText("LOADING....")).not.toBeInTheDocument();
  });
});