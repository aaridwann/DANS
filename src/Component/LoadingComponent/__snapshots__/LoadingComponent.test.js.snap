// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent does not render when loading state is false 1`] = `null`;

exports[`LoadingComponent renders correctly when loading state is true 1`] = `
<div
  class="absolute flex flex-col z-[500] items-center mx-auto justify-center bg-white/50 w-screen h-screen p-4"
>
  <img
    class=" w-20 h-auto mb-4"
    src="mocked-image"
  />
  <h1
    class=" text-2xl text-cyan-600 shadow-cyan-800  drop-shadow-2xl"
  >
    LOADING....
  </h1>
</div>
`;
